app/kumactl/cmd/delete/delete.go

```diff
@@ -1,23 +1,64 @@
 package delete
 
 import (
+	"context"
 	kumactl_cmd "github.com/Kong/kuma/app/kumactl/pkg/cmd"
+	"github.com/Kong/kuma/pkg/core/resources/model"
+	"github.com/Kong/kuma/pkg/core/resources/registry"
+	"github.com/Kong/kuma/pkg/core/resources/store"
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
-type deleteContext struct {
-	*kumactl_cmd.RootContext
-}
-
 func NewDeleteCmd(pctx *kumactl_cmd.RootContext) *cobra.Command {
-	ctx := &deleteContext{RootContext: pctx}
 	cmd := &cobra.Command{
 		Use:   "delete",
 		Short: "Delete Kuma resources",
 		Long:  `Delete Kuma resources.`,
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			rs, err := pctx.CurrentResourceStore()
+			if err != nil {
+				return err
+			}
+			resourceTypeArg := args[0]
+			name := args[1]
+
+			var resource model.Resource
+			var resourceType model.ResourceType
+			switch resourceTypeArg {
+			case "mesh":
+				resourceType = model.ResourceType("Mesh")
```

Comment:
Let's reuse predefined constants for resource types, e.g. `mesh.MeshType` (see https://github.com/Kong/kuma/blob/d859006/pkg/core/resources/apis/mesh/mesh.go#L11)